### DISCUSSION TITLE
Combine claimable balances with path-payment

### DISCUSSION THREAD

**Discussion by hanseartic** [kalepail/stellar-quest-bounties#77] [Bounty Ideas]
# Scenario
When some account is added as claimant to a claimable balance they can (from the API perspective) decide to claim or not. In order to claim a given asset a trustline to that asset needs to exist in the claimant account. In some cases it may be desirable to swap the claimable asset to another asset during claiming and not add another trustline.

* An account holds an anchored asset or stablecoin (e.g. USDC) and is added as a claimant to claim another asset representing the same entity (e.g. USDT) but they don't want to deal with different representations.
* An account is eligible for a reward-asset (think reviewer credits) but they'd rather cash-out immediately.

## Possible bounty
Building – or adding to an existing – service functionality to actually choose a destination asset for a claimable balance.

* The user should be presented with the estimated amount of target asset
* The user should be able to sign and submit the transaction without dealing with raw key (i.e. integration of wallet / extensions like freighter, albedo or even lobstr vault is a **must**)

**Comment by kalepail** [comment]
Seems like a great educational / continuous bounty. It'd need to be scoped simply but I like the general concept. A trustline would be needed but could be ephemeral and atomic in a larger transaction. 

I like it!
